src/rust/src/wallet_tool.rs

```diff
@@ -0,0 +1,522 @@
+        println!(concat!(
+            "\nIt looks like zcashd is running without the '-exportdir' option.\n\n",
+            "Please start or restart zcashd with '-exportdir' set to the absolute\n",
+            "path of the directory you want to save the wallet export file to.\n",
+            "(Don't forget to restart zcashd without '-exportdir' after finishing\n",
+            "the backup, if running it long-term with that option is not desired\n",
+            "or would be a security hazard in your environment.)"
+        ));
+        return Err(io::Error::from(ErrorKind::Other));
+    }
+    println!("Yes, and it is running with the '-exportdir' option as required.");
+
+    let mut stdin = io::stdin();
+
+    let base = default_filename_base();
+    let mut r = 0u32;
+    let out = loop {
+        let default_filename = if r != 0 {
+            format!("{}r{}", base, r)
+        } else {
+            base.to_string()
+        };
+        println!(
+            concat!(
+                "\nEnter the filename for the wallet export file, using only characters\n",
+                "a-z, A-Z and 0-9 (default '{}')."
+            ),
+            default_filename
+        );
+        let mut buf = String::new();
+        let response = prompt(&mut stdin, &mut buf)?;
+        let filename = if response.is_empty() {
+            r = r.saturating_add(1);
+            &default_filename
+        } else {
+            response
+        };
+        if opts.debug {
+            eprintln!("DEBUG: Using filename {:?}", filename);
+        }
+
+        let mut cli_args = cli_options.clone();
+        cli_args.extend_from_slice(&["z_exportwallet".to_string(), filename.to_string()]);
+        let out = exec(&zcash_cli, &cli_args, opts.debug)?;
+        let cli_err: Vec<_> = from_utf8(&out.stderr)
+            .map_err(|_| io::Error::from(ErrorKind::InvalidData))?
+            .lines()
+            .map(|s| s.trim_end_matches('\r'))
+            .collect();
+        if opts.debug {
+            eprintln!("DEBUG: stderr {:?}", cli_err);
+        }
+        if cli_err.len() >= 3
+            && cli_err[0] == "error code: -8"
+            && cli_err[1] == "error message:"
+            && cli_err[2].contains("overwrite existing")
+        {
+            println!(concat!(
+                "That file already exists. Please pick a unique filename in the\n",
+                "directory specified by the '-exportdir' option to zcashd."
+            ));
+            continue;
+        } else {
+            break out;
+        }
+    };
+
+    let cli_out: Vec<_> = from_utf8(&out.stdout)
+        .map_err(|_| io::Error::from(ErrorKind::InvalidData))?
+        .lines()
+        .map(|s| s.trim_end_matches('\r'))
+        .collect();
+    if opts.debug {
+        eprintln!("DEBUG: stdout {:?}", cli_out);
+    }
+    if cli_out.is_empty() {
+        return Err(io::Error::from(ErrorKind::InvalidData));
+    }
+    let export_path = cli_out[0].trim_matches('"');
```

Comment:
Oh, I guess this is a JSON string and should be unescaped.

Reply:
Actually it shouldn't; this is the output from `zcash-cli` which is already unescaped. Also don't need to strip double quotes (technically, double quotes are valid characters in a Unix filename).